README.md

```diff
@@ -8,35 +8,88 @@ The main command in this project is `wemmick` which has many uses:
 - Create suites from JSON Schema files
 - Run any existing expectations suite against any table in any existing datasource
 
-## Installation
+This repo provides the library which can be run within a local python environment and docker images for portable runtimes.
```

Comment:
the same cli lib will be the http interface? 

how about modes of operation? 

WEMMICK_MODE=CLI (the default)  we use the cli interface as described below
WEMMICK_MODE=HTTP  accepts incoming over the simple rest api 

When/if we add gRPC interface we can accept WEMMICK_MODE=GRPC  ( Even though we can get a typed sdk from OpenAPI, Im biased for the gRPC because it models the functional command & control usage whereas REST is perfect for representing a declarative state)

Reply:
1. The cli library is not the same as the http interface. There is zero coupling between them, though there is a little duplication I could easily remove which would make them coupled. Therefore I'm not sure modes makes sense, though I could be convinced otherwise.

2. I have no strong opinion for HTTP over gRPC other than the http server was super quick to implement (top of mind vs looking up grpc libraries). Keep in mind the HTTP server only functions to trigger validations via Cloud Scheduler (which cannot invoke a container directly).

Reply:
This diff sentence is intended to highlight that a person could use the library/cli in their own python env or in a portable runtime.

Reply:
1. I follow the two are just different wrappers/helpers and independent of each other - but they get bundled in the same wheel right?  I like the same docker image that can be used for the cli or http. Rather than router entry point that looks at the mode env var - the service's infra as code can just override the entrypoint, no extra work there.

2. ya fastapi looks way slick! dont need to address grpc in this design.  treat my comment above as more about the one wemmick docker image suggestion

Reply:
1. Now we are getting somewhere - a single wemmick image with overridable entrypoints! That is very slick. I'll do that.
2. Roger that.

Reply:
Design doc updated to reflect this!